nautobot/extras/jobs.py

```diff
@@ -993,7 +1000,7 @@ def run_job(data, request, job_result_pk, commit=True, *args, **kwargs):
 
```

Comment:
I'm also concerned about the line above at line 987/994:

```python
file_ids = [data[f] for f in file_fields]
```

Seems like if `data` isn't a dict or any referenced `file_field`is omitted from `data`, this would be another case where we could error out without it being recorded in the JobResult. Perhaps a fix might be to simply move up the below `try:` so that the above logic is included in the try/except block as well?

Reply:
My only concern here is that there are several steps now all wrapped here in a `try...except` block. Even if it's more verbose, we might want to think about deliberate exception handling for each step? Not to overcomplicate, but perhaps to make the error-handling more meaningful? Feel free to disregard if you think I'm overthinking it. :)

Reply:
I'm not sure how readable it would be to have a whole series of separate try/except blocks instead - fundamentally we do need to be able to catch any plausible exception at each step so as to ensure that we leave the JobResult in an appropriate state and do any necessary cleanup.